[Java] Checkstyle cannot initialize module TreeWalker

Thank you for reporting the problem!
Please make sure what you are reporting is a bug with reproducible steps. 

**Describe the bug**
Checkstyle can't run:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (default-cli) on project fury-parent: Failed during checkstyle configuration: cannot initialize module TreeWalker - TreeWalker is not allowed as a parent of LineLength Please review 'Parent Module' section for this Check in web documentation if Check is standard. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
(py3.7) chaokunyang@B-M856MD6M-2323 java % git push ant

```
Maybe related to #592

**Expected behavior**
A clear and concise description of what you expected to happen.

**To Reproduce**
```bash
cd java
mvn checkstyle:check
```

**Screenshots**
![image](https://github.com/alipay/fury/assets/12445254/3d1cbc65-4423-4387-94dc-b6c1060c1e3a)
**Environment (please complete the following information):**
Please specify the versions of Fury, OS, and others such as JDK/Python/Gcc/Clang/Go/NodeJS if they are used.

--- a/java/checkstyle.xml
+++ b/java/checkstyle.xml
@@ -46,6 +46,11 @@
     <property name="fileNamePattern"
               value="io.fury.*MurmurHash3|Identity2IdMap.java"/>
   </module>
+  <module name="LineLength">
+    <property name="max" value="120"/>
+    <property name="ignorePattern"
+              value="^package.*|^import.*|a href|href|http://|https://|ftp://"/>
+  </module>
   <module name="TreeWalker">
     <module name="SuppressWarningsHolder" />
     <module name="SuppressionCommentFilter">
@@ -68,11 +73,6 @@
       <property name="allowEscapesForControlCharacters" value="true"/>
       <property name="allowByTailComment" value="true"/>
       <property name="allowNonPrintableEscapes" value="true"/>
-    </module>
-    <module name="LineLength">
-      <property name="max" value="120"/>
-      <property name="ignorePattern"
-                value="^package.*|^import.*|a href|href|http://|https://|ftp://"/>
     </module>
     <module name="AvoidStarImport"/>
     <module name="NoLineWrap"/>
@@ -223,12 +223,8 @@
     <module name="JavadocMethod">
       <property name="scope" value="public"/>
       <property name="allowMissingParamTags" value="true"/>
-      <property name="allowMissingThrowsTags" value="true"/>
       <property name="allowMissingReturnTag" value="true"/>
-      <property name="allowMissingJavadoc" value="true"/>
-      <property name="minLineCount" value="2"/>
       <property name="allowedAnnotations" value="Override, Test"/>
-      <property name="allowThrowsTagsForSubclasses" value="true"/>
     </module>
     <module name="MethodName">
       <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9_]*$"/>

--- a/java/fury-core/src/main/java/io/fury/resolver/RefResolver.java
+++ b/java/fury-core/src/main/java/io/fury/resolver/RefResolver.java
@@ -27,7 +27,6 @@ import io.fury.memory.MemoryBuffer;
  * @author chaokunyang
  */
 public interface RefResolver {
-  // TODO(mubai) add putIfAbsent for cuckoo hash.
   /**
    * Write reference and tag for the obj if the obj has been written previously, write null/not-null
    * tag otherwise.

--- a/java/fury-core/src/main/java/io/fury/util/unsafe/_JDKAccess.java
+++ b/java/fury-core/src/main/java/io/fury/util/unsafe/_JDKAccess.java
@@ -11,7 +11,9 @@ import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import sun.misc.Unsafe;
 
+// CHECKSTYLE.OFF:TypeName
 public class _JDKAccess {
+  // CHECKSTYLE.OFF:TypeName
   public static final int JAVA_VERSION;
   public static final boolean OPEN_J9;
   public static final Unsafe UNSAFE;
@@ -51,7 +53,10 @@ public class _JDKAccess {
     }
   }
 
+  // CHECKSTYLE.OFF:MethodName
+
   public static Lookup _trustedLookup(Class<?> objectClass) {
+    // CHECKSTYLE.ON:MethodName
     return _Lookup._trustedLookup(objectClass);
   }
 

--- a/java/fury-core/src/main/java/io/fury/util/unsafe/_Lookup.java
+++ b/java/fury-core/src/main/java/io/fury/util/unsafe/_Lookup.java
@@ -7,7 +7,9 @@ import java.lang.invoke.MethodType;
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 
+// CHECKSTYLE.OFF:TypeName
 class _Lookup {
+  // CHECKSTYLE.ON:TypeName
   static final Lookup IMPL_LOOKUP;
   static volatile MethodHandle CONSTRUCTOR_LOOKUP;
   static volatile boolean CONSTRUCTOR_LOOKUP_ERROR;
@@ -31,10 +33,12 @@ class _Lookup {
     }
   }
 
+  // CHECKSTYLE.OFF:MethodName
   public static Lookup _trustedLookup(Class<?> objectClass) {
+    // CHECKSTYLE.OFF:MethodName
     if (!CONSTRUCTOR_LOOKUP_ERROR) {
       try {
-        int TRUSTED = -1;
+        int trusted = -1;
         MethodHandle constructor = CONSTRUCTOR_LOOKUP;
         if (_JDKAccess.JAVA_VERSION < 15) {
           if (constructor == null) {
@@ -43,9 +47,9 @@ class _Lookup {
                     Lookup.class, MethodType.methodType(void.class, Class.class, int.class));
             CONSTRUCTOR_LOOKUP = constructor;
           }
-          int FULL_ACCESS_MASK = 31; // for IBM Open J9 JDK
+          int fullAccessMask = 31; // for IBM Open J9 JDK
           return (Lookup)
-              constructor.invoke(objectClass, _JDKAccess.OPEN_J9 ? FULL_ACCESS_MASK : TRUSTED);
+              constructor.invoke(objectClass, _JDKAccess.OPEN_J9 ? fullAccessMask : trusted);
         } else {
           if (constructor == null) {
             constructor =
@@ -54,7 +58,7 @@ class _Lookup {
                     MethodType.methodType(void.class, Class.class, Class.class, int.class));
             CONSTRUCTOR_LOOKUP = constructor;
           }
-          return (Lookup) constructor.invoke(objectClass, null, TRUSTED);
+          return (Lookup) constructor.invoke(objectClass, null, trusted);
         }
       } catch (Throwable ignored) {
         CONSTRUCTOR_LOOKUP_ERROR = true;

--- a/java/fury-test-core/src/main/java/io/fury/test/bean/BeanB.java
+++ b/java/fury-test-core/src/main/java/io/fury/test/bean/BeanB.java
@@ -34,6 +34,7 @@ public final class BeanB implements Serializable {
   private int[] intArr;
   private List<Integer> intList;
 
+  /** Create Object. */
   public static BeanB createBeanB(int arrSize) {
     Random rnd = new Random(37);
     BeanB beanB = new BeanB();

--- a/java/fury-test-core/src/main/java/io/fury/test/bean/CollectionFields.java
+++ b/java/fury-test-core/src/main/java/io/fury/test/bean/CollectionFields.java
@@ -67,6 +67,7 @@ public class CollectionFields {
     return (CollectionFields) copyToCanEqual(this, new CollectionFields());
   }
 
+  /** Copy the object, so that inner fields are equal able. */
   public static Object copyToCanEqual(Object raw, Object newObj) {
     for (Field field : raw.getClass().getDeclaredFields()) {
       field.setAccessible(true);

--- a/java/fury-test-core/src/main/java/io/fury/test/bean/Cyclic.java
+++ b/java/fury-test-core/src/main/java/io/fury/test/bean/Cyclic.java
@@ -50,6 +50,7 @@ public class Cyclic {
     }
   }
 
+  /** Create Object. */
   public static Cyclic create(boolean circular) {
     Cyclic cyclic = new Cyclic();
     cyclic.f1 = "str";

--- a/java/fury-test-core/src/main/java/io/fury/test/bean/FinalCyclic.java
+++ b/java/fury-test-core/src/main/java/io/fury/test/bean/FinalCyclic.java
@@ -50,6 +50,7 @@ public final class FinalCyclic {
     }
   }
 
+  /** Create Object. */
   public static FinalCyclic create(boolean circular) {
     FinalCyclic cyclic = new FinalCyclic();
     cyclic.f1 = "str";

--- a/java/fury-test-core/src/main/java/io/fury/test/bean/Foo.java
+++ b/java/fury-test-core/src/main/java/io/fury/test/bean/Foo.java
@@ -46,6 +46,7 @@ public class Foo implements Serializable {
   long f14;
   long f15;
 
+  /** Create Object. */
   public static Foo create() {
     Random random = new Random(31);
     Foo foo = new Foo();
@@ -67,6 +68,7 @@ public class Foo implements Serializable {
     return foo;
   }
 
+  /** Create Class. */
   public static Class<?> createCompatibleClass1() {
     String pkg = Foo.class.getPackage().getName();
     String code =
@@ -84,6 +86,7 @@ public class Foo implements Serializable {
     return loadFooClass(pkg, code);
   }
 
+  /** Create class. */
   public static Class<?> createCompatibleClass2() {
     String pkg = Foo.class.getPackage().getName();
     String code =
@@ -99,6 +102,7 @@ public class Foo implements Serializable {
     return loadFooClass(pkg, code);
   }
 
+  /** Create class. */
   public static Class<?> createCompatibleClass3() {
     String pkg = Foo.class.getPackage().getName();
     String code =

--- a/java/fury-test-core/src/main/java/io/fury/test/bean/Struct.java
+++ b/java/fury-test-core/src/main/java/io/fury/test/bean/Struct.java
@@ -47,6 +47,7 @@ import org.apache.commons.lang3.StringUtils;
  */
 public final class Struct implements Serializable {
 
+  /** Return object string repr. */
   public static String toString(Object o) {
     StringBuilder builder = new StringBuilder(o.getClass().getSimpleName() + "(");
     try {
@@ -71,6 +72,7 @@ public final class Struct implements Serializable {
     return builder.toString();
   }
 
+  /** Return whether tow objects equals. */
   public static boolean equalsWith(Object o1, Object o2) {
     if (o1 == o2) {
       return true;
@@ -119,6 +121,7 @@ public final class Struct implements Serializable {
     return true;
   }
 
+  /** Create Object. */
   public static Object create(String classname, int repeat) {
     try {
       return createPOJO(Struct.createStructClass(classname, repeat));
@@ -127,6 +130,7 @@ public final class Struct implements Serializable {
     }
   }
 
+  /** Create Object. */
   public static Object createPOJO(Class<?> clz) {
     Random random = new Random(17);
     try {
@@ -181,6 +185,7 @@ public final class Struct implements Serializable {
     }
   }
 
+  /** Create class. */
   public static Class<?> createNumberStructClass(String classname, int repeat) {
     if (StringUtils.isBlank(classname)) {
       classname = String.format("Struct%d", System.currentTimeMillis());
@@ -221,6 +226,7 @@ public final class Struct implements Serializable {
     return createStructClass(classname, classCode.toString());
   }
 
+  /** Create Class. */
   public static Class<?> createStructClass(String classname, int repeat) {
     if (StringUtils.isBlank(classname)) {
       classname = String.format("Struct%d", System.currentTimeMillis());
@@ -263,6 +269,7 @@ public final class Struct implements Serializable {
     return createStructClass(classname, classCode.toString());
   }
 
+  /** Create class. */
   public static Class<?> createStructClass(String classname, String classCode) {
 
     Path path = Paths.get(classname + ".java");

